dartagnan/src/main/java/com/dat3m/dartagnan/verification/model/event/DefaultEventModel.java

```diff
@@ -0,0 +1,99 @@
+package com.dat3m.dartagnan.verification.model.event;
+
+import com.dat3m.dartagnan.program.event.core.*;
+import com.dat3m.dartagnan.program.event.Event;
+import com.dat3m.dartagnan.program.event.RegReader;
+import com.dat3m.dartagnan.program.event.RegWriter;
+import com.dat3m.dartagnan.verification.model.ThreadModel;
+
+import static com.dat3m.dartagnan.program.event.Tag.*;
+
+
+public class DefaultEventModel implements EventModel {
+    protected final Event event;
+    protected final ThreadModel thread;
```

Comment:
This again gives me the idea that the design is ugly. We are keeping track of the same information in different places, e.g. the ThreadModel referring to its events and the eventModels referring to their threadModel

Reply:
I don't think the cyclic reference is necessarily bad. Events knowing their threads and threads knowing their events is fine in principle.